tests/test_smoke_run_recover_stuck_done.out

```diff
@@ -0,0 +1,49 @@
+recovering from log: -
+recovered state
++ saga execution: 049b2522-308d-442e-bc65-9bfaef863597
++-- done: (start node)
++-- undo-failed: InstanceCreate (produces "instance_id")
++-- undone: (constant = {"number_of_things":1}) (produces "server_alloc_params")
++-- (parallel actions):
+        +-- undone: VpcAllocIp (produces "instance_ip")
+        +-- undone: VolumeCreate (produces "volume_id")
+        +-- undone: (subsaga start: "server-alloc")
+                +-- undone: ServerPick (produces "server_id")
+                +-- undone: ServerReserve (produces "server_reserve")
+        +-- undone: (subsaga end) (produces "server_alloc")
++-- undone: InstanceConfigure (produces "instance_configure")
++-- undone: VolumeAttach (produces "volume_attach")
++-- failed: InstanceBoot (produces "instance_boot")
```

Comment:
Just to clarify my understanding here: If `InstanceBoot` fails because of an undo failure, the rest of the actions above it won't actually be undone, right? I got confused, because they are all marked with "undone".

Reply:
Good question.  In this example, InstanceBoot failed because of an _action_ failure.  Then, while unwinding, InstanceCreate's undo action failed.  So those middle ones (like VolumeAttach) _will_ have been undone.

Reply:
Oh I see. Thanks!